### PR TITLE
Med 67 implement local doctor navigation UI

### PR DESCRIPTION
**Pull request by SoulOfDarknes**
Add navigation sidebar

1. add colors animations
2. routs and links
3. placeholder for name and foto

**PR checklist:**

**1.1 Common**
- [x] 1.1.2 no any, should be also added to eslint rules
- [x] 1.1.1 types for input and output params
- [x] 1.1.3 try/catch for any async function
- [x] 1.1.4 no magic numbers
- [x] 1.1.5 compare only with constants not with strings (instead of user === ‘admin’, better user === roles.admin)
- [x] 1.1.6 no ternary operator inside ternary operator (bad example: question ? first: second ? third: fourth)
- [x] 1.1.7 avoid similar types with duplicating by generics
- [x] 1.1.8 no console.log in pr
- [x] 1.1.9 .env file should be in .gitignore
- [x] 1.1.10 delete commented code if it's not part of documentation
- [x] 1.1.11 no links in the code, env links should be in env file (for example: server url),
- [x] constant links (for example default avatar URL) should be in constant file
- [x] 1.1.12 constants and function names should be lowercase.
- [x] 1.1.13 no dates format in the code (like "dd/MM/yyyy”), move it in global constant variable
- [x] 1.1.14 import rules. imports should come in a specific order: node modules, absolute path, relative path
- [x] 1.1.15 strings should be in the constant variable. Example: instead of ‘15 3 * * *’, const EACH_DAY_15h_15min = ‘15 3 * * *’

 **1.3 Frontend|Mobile**
- [x] 1.3.1 split component and logic (move logic to custom hook file)
- [x] 1.3.2 colors, font size, and font name should be in the theme or in the constants
- [x] 1.3.3 no text in the components, use i18n approach, even if you have only one language for now
- [x] 1.3.4 inline styles prohibited
- [x] 1.3.5 import should be absolute. instead of ../../../components/myComponent should be components/myComponent

image (9)

### CODE REVIEW

File: apps/medon-fe/src/components/LoginComponent/index.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import Logo from 'components/Logo';
+import { Title, Text } from 'components/LoginComponent/style';
+
+
+
+export default function LoginComponent() {
+
+  const handleSubmit = (data: LoginFormValues) =>  data
+  
+
```
Comment (lyudad) [inline apps/medon-fe/src/components/LoginComponent/index.tsx]: why this space needed?

File: apps/medon-fe/src/components/LoginComponent/style.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import styled from "styled-components";
+import { theme } from "styles/theme";
+
+
+export const Title = styled.h1`
+  color: ${theme.colors.title};
+  font-family: ${theme.typography.fontFamily.roboto};
+  font-style: normal;
+  font-weight: 700;
+  font-size: 2rem;
```
Comment (lyudad) [inline apps/medon-fe/src/components/LoginComponent/style.tsx]: to constant

File: apps/medon-fe/src/components/LoginForm/index.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+import {
+  StyledErrorMessage,
+  Form,
+  DontHaveButton,
+  ForgotButton,
+  SendButton,
+} from 'components/LoginForm/style';
+import { loginFormSchema } from 'components/FormSchema/index';
+
+
```
Comment (lyudad) [inline apps/medon-fe/src/components/LoginForm/index.tsx]: why this space needed?

File: apps/medon-fe/src/components/LoginForm/style.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      text-align: left;
+      position: relative;
+    }
+
+`;
+
+export const SendButton = styled(Input)`
+      background: ${theme.colors.blue_300};
+      color: ${theme.colors.white};
+      cursor: pointer;
```
Comment (lyudad) [inline apps/medon-fe/src/components/LoginForm/style.tsx]: you have problem with formating

File: apps/medon-fe/src/components/LoginComponent/index.tsx (hunk 1 above)
Comment (SoulOfDarknes) [reply apps/medon-fe/src/components/LoginComponent/index.tsx]: fixed

File: apps/medon-fe/src/components/LoginComponent/style.tsx (hunk 2 above)
Comment (SoulOfDarknes) [reply apps/medon-fe/src/components/LoginComponent/style.tsx]: added

File: apps/medon-fe/src/components/LoginForm/style.tsx (hunk 4 above)
Comment (SoulOfDarknes) [reply apps/medon-fe/src/components/LoginForm/style.tsx]: fixed

File: apps/medon-fe/src/components/LoginForm/index.tsx (hunk 3 above)
Comment (SoulOfDarknes) [reply apps/medon-fe/src/components/LoginForm/index.tsx]: fixed

File: apps/medon-fe/src/assets/images/Avatar.svg
Comment (lyudad) [inline apps/medon-fe/src/assets/images/Avatar.svg:1]: its bad example of default image

File: apps/medon-fe/src/components/FormSchema/index.tsx [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import * as Yup from "yup";
+
+export const loginFormSchema = Yup.object({
+  email: Yup.string().email().required("E-mail is required"),
+  password: Yup.string()
+    .min(6, "Password must be at least 6 characters long")
```
Comment (lyudad) [inline apps/medon-fe/src/components/FormSchema/index.tsx]: should be in translation file

File: apps/medon-fe/src/components/Navigation/styles.tsx [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+export const Links = styled(Link)`
+  text-decoration: none;
+  color: ${theme.colors.navi_text};
+  font-family: ${theme.fontFamily.roboto};
+  font-weight: 700;
+      &:hover {
+        color: ${theme.colors.icon_active};
+      }
+`
+
```
Comment (lyudad) [inline apps/medon-fe/src/components/Navigation/styles.tsx:36]: why this space needed?

File: apps/medon-fe/src/components/Navigation/styles.tsx [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+  border-radius: 50%;
+  margin-right: 8px;
+`;
+
+export const UserName = styled.span`
+  font-size: ${theme.fontSizes.md};
+  font-weight: 500;
+  color: ${theme.colors.black};
+  font-family: ${theme.fontFamily.sf_pro_text};
+
```
Comment (lyudad) [inline apps/medon-fe/src/components/Navigation/styles.tsx]: you have issue with formating, why this space needed

Comment (SoulOfDarknes) [reply apps/medon-fe/src/components/Navigation/styles.tsx]: fix my eslint

File: apps/medon-fe/src/components/Navigation/styles.tsx (hunk 6 above)
Comment (SoulOfDarknes) [reply apps/medon-fe/src/components/Navigation/styles.tsx:36]: fix my eslint

File: apps/medon-fe/src/assets/images/Avatar.svg
Comment (SoulOfDarknes) [reply apps/medon-fe/src/assets/images/Avatar.svg:1]: changed

File: apps/medon-fe/src/components/FormSchema/index.tsx (hunk 5 above)
Comment (SoulOfDarknes) [reply apps/medon-fe/src/components/FormSchema/index.tsx]: fixed

File: apps/medon-fe/src/components/Navigation/index.tsx [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+  UserBlock,
+  UserAvatar,
+  UserName,
+} from 'components/Navigation/styles';
+import profileImagePlaceholder from 'assets/images/Avatar.svg';
+import Logo from 'components/Logo';
+
+export default function Navigation() {
+  const { t } = useTranslation();
+  const storedUser = localStorage.getItem('user');
```
Comment (lyudad) [inline apps/medon-fe/src/components/Navigation/index.tsx]: no, redux-persist

File: apps/medon-fe/src/components/Navigation/index.tsx [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+  const user = storedUser ? JSON.parse(storedUser) : null;
+
+  const navItems = [
+    { to: '/dashboard', icon: <Dashboard />, label: t('navigation.dashboard') },
```
Comment (lyudad) [inline apps/medon-fe/src/components/Navigation/index.tsx]: `to` should be in constants

File: apps/medon-fe/src/components/Navigation/index.tsx (hunk 8 above)
Comment (SoulOfDarknes) [reply apps/medon-fe/src/components/Navigation/index.tsx]: changed

File: apps/medon-fe/src/components/Navigation/index.tsx (hunk 9 above)
Comment (SoulOfDarknes) [reply apps/medon-fe/src/components/Navigation/index.tsx]: fixed

File: apps/medon-fe/src/components/LoginForm/index.tsx [hunk 10]
```diff
@@ -12,34 +12,40 @@ import {
...
   });
 
-  const handleFormSubmit = handleSubmit((data) => {
-    onSubmit(data);
-  });
+  const handleFormSubmit = async (formData: LoginRequest) => {
+    try {
+      await login(formData).unwrap();
+    } catch (err) {
+      console.error('Failed to login', err);
```
Comment (skazzp) [inline apps/medon-fe/src/components/LoginForm/index.tsx]: change console.log to toastify

**Review by skazzp — CHANGES_REQUESTED** [review]

**Review by Mati-Pereira — APPROVED** [review]
Just remove the console.log

**Review by skazzp — APPROVED** [review]

File: apps/medon-fe/src/components/Navigation/index.tsx [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+import Logo from 'components/Logo';
+import { useDispatch } from 'react-redux';
+import { useAppSelector } from 'redux/hooks';
+import { getUserSelector } from 'redux/features/userSlice/userSelectors';
+import {
+  initialState,
+  setToken,
+  setUser,
+} from 'redux/features/userSlice/userSlice';
+import {navigation} from 'utils/constants/navigation'
```
Comment (lyudad) [inline apps/medon-fe/src/components/Navigation/index.tsx]: you dont configure prettier. compare with line 20

File: apps/medon-fe/src/components/Navigation/index.tsx [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+                {icon}
+                <div>{t(label)}</div>
+              </li>
+            </NavLinkStyled>
+          ))}
+        </Ul>
+        <UserBlock>
+          <UserAvatar
+            src={user.photo || profileImagePlaceholder}
+            alt="User avatar"
```
Comment (lyudad) [inline apps/medon-fe/src/components/Navigation/index.tsx]: to text file

File: apps/medon-fe/src/components/Navigation/styles.tsx [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+import { ReactComponent as DashboardIcon } from 'assets/images/navigation/Elements.svg';
+import { ReactComponent as BriefcaseIcon } from 'assets/images/navigation/Briefcase.svg';
+import { ReactComponent as ProfileIcon } from 'assets/images/navigation/User_Circle.svg';
+import { ReactComponent as PatientIcon } from 'assets/images/navigation/Users.svg';
+import { ReactComponent as HelpIcon } from 'assets/images/navigation/Question.svg';
+import { ReactComponent as LogoutIcon } from 'assets/images/navigation/Sign_Out.svg';
+
+import { theme } from 'styles/theme';
+
+
```
Comment (lyudad) [inline apps/medon-fe/src/components/Navigation/styles.tsx]: why this empty line neded?

File: apps/medon-fe/src/components/Navigation/index.tsx (hunk 11 above)
Comment (SoulOfDarknes) [reply apps/medon-fe/src/components/Navigation/index.tsx]: fixed

File: apps/medon-fe/src/components/Navigation/styles.tsx (hunk 13 above)
Comment (SoulOfDarknes) [reply apps/medon-fe/src/components/Navigation/styles.tsx]: fixed

File: apps/medon-fe/src/components/Navigation/index.tsx (hunk 12 above)
Comment (SoulOfDarknes) [reply apps/medon-fe/src/components/Navigation/index.tsx]: added

**Pull request merged**